pkg/gcp/machine_controller_util.go

```diff
@@ -83,32 +85,27 @@ func (ms *MachinePlugin) CreateMachineUtil(ctx context.Context, machineName stri
 		instance.Description = *providerSpec.Description
 	}
 
-	disks := []*compute.AttachedDisk{}
+	var disks []*compute.AttachedDisk
 	for _, disk := range providerSpec.Disks {
-		var attachedDisk compute.AttachedDisk
-		autoDelete := false
-		if disk.AutoDelete == nil || *disk.AutoDelete == true {
-			autoDelete = true
+		attachedDisk := compute.AttachedDisk{
+			Type:       validation.DiskTypePersistent,
+			Boot:       disk.Boot,
+			AutoDelete: ptr.Deref(disk.AutoDelete, true),
+			InitializeParams: &compute.AttachedDiskInitializeParams{
+				DiskSizeGb:            disk.SizeGb,
+				DiskType:              fmt.Sprintf("zones/%s/diskTypes/%s", zone, disk.Type),
+				Labels:                disk.Labels,
+				SourceImage:           disk.Image,
+				ProvisionedIops:       disk.ProvisionedIops,
+				ProvisionedThroughput: disk.ProvisionedThroughput,
+			},
 		}
 		if disk.Type == validation.DiskTypeScratch {
-			attachedDisk = compute.AttachedDisk{
-				AutoDelete: autoDelete,
-				Type:       disk.Type,
-				Interface:  disk.Interface,
-				InitializeParams: &compute.AttachedDiskInitializeParams{
-					DiskType: fmt.Sprintf("zones/%s/diskTypes/%s", zone, "local-ssd"),
-				},
-			}
-		} else {
-			attachedDisk = compute.AttachedDisk{
-				AutoDelete: autoDelete,
-				Boot:       disk.Boot,
-				InitializeParams: &compute.AttachedDiskInitializeParams{
-					DiskSizeGb:  disk.SizeGb,
-					DiskType:    fmt.Sprintf("zones/%s/diskTypes/%s", zone, disk.Type),
-					Labels:      disk.Labels,
-					SourceImage: disk.Image,
-				},
+			attachedDisk.Type = validation.DiskTypeScratch
+			attachedDisk.Boot = false
+			attachedDisk.Interface = disk.Interface
+			attachedDisk.InitializeParams = &compute.AttachedDiskInitializeParams{
+				DiskType: fmt.Sprintf("zones/%s/diskTypes/%s", zone, "local-ssd"),
```

Comment:
minor nit: I would maybe consider replacing the whole struct instead of doing the assignments here because the end-result is kinda harder to read (must mentally merge the 2)